spec/api/v1/sg_email_spec.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+# rubocop:disable Layout/AlignHash
+
+require 'rails_helper'
+
+RSpec.describe 'SendGrid API', type: :request do
+  describe 'POST /api/v1/sg_emails' do
+    let(:valid_attributes) do
+      {
+        'from_email': 'alice@gmail.com',
+        'to_email':   'bob@gmail.com',
+        'subject':    'A test from Rails',
+        'content':    'How R U?'
+      }.to_json
+    end
+
+    it 'get all messages' do
+      uri = URI(ENV['API_ENDPOINT'] + 'sg_emails')
+      response = JSON.parse(Net::HTTP.get(uri))
+      expect(response['all'].first['id']).to eq 1
+      expect(response['all'].first['response_status_code']).to eq '202'
+    end
+
+    it 'send a message' do
+      uri = URI(ENV['API_ENDPOINT'] + 'sg_emails')
+      # Get request data from fixture
+      form_data = JSON.parse(get_sendgrid_request_data('post_sg_email'))
+
+      # Post request to fake clearstream server
+      #response = Net::HTTP.post_form(uri, form_data)
+      request = Net::HTTP::Post.new(uri.request_uri)
+      request.set_form_data(form_data)
+      request["User-Agent"] = "Ruby Spec"
+      request["Accept"] = "*/*"
+      request["Content-Type"] = "application/json"
+      request['Authorization'] = 'Basic ' + ENV['ADMIN_BASE64_CREDS']
+      http = Net::HTTP.new(uri.host, uri.port)
+      response = http.request(request)
+
+      response = JSON.parse(response.body)
+      expect(response['status']).to eq '202'
+    end
+
+
+
+    # context 'when request is valid' do
```

Comment:
If we are not using it, it's better to remove this commented code.

Reply:
Let's clean up the tests in the next feature branch, okay?